server/lib/tuist_web/controllers/api/auth_controller.ex

```diff
@@ -160,16 +161,23 @@ defmodule TuistWeb.API.AuthController do
         |> put_status(:unauthorized)
         |> json(%{message: "The refresh token is expired or invalid"})
 
-      {:ok, _old_token, {new_refresh_token, _new_claims}} ->
-        {:ok, _old_token_with_claims, {new_access_token, _new_access_token_claims}} =
-          Authentication.exchange(new_refresh_token, "refresh", "access", ttl: @access_token_ttl)
+      {:ok, {_old_token, old_claims}, {new_refresh_token, _new_claims}} ->
+        case Authentication.exchange(new_refresh_token, "refresh", "access", ttl: @access_token_ttl) do
+          {:ok, _old_token_with_claims, {new_access_token, _new_access_token_claims}} ->
+            conn
+            |> put_status(:ok)
+            |> json(%{
+              access_token: new_access_token,
+              refresh_token: new_refresh_token
+            })
 
-        conn
-        |> put_status(:ok)
-        |> json(%{
-          access_token: new_access_token,
-          refresh_token: new_refresh_token
-        })
+          {:error, :invalid_token_type} ->
+            conn
+            |> put_status(:bad_request)
+            |> json(%{
+              message: "The token can't be refreshed because it has invalid type: #{Map.get(old_claims, "typ")}"
+            })
```

Comment:
won't this limit our visibility into when this happens? these errors will no longer appear on AppSignal, but users will still get logged out.

Reply:
It shifts it to the client. I added an error reporting which I expect to be temporary until we figure out if it's a bug from us or not.

Reply:
right, but we don't have observability in the client. Anyways, thanks for addressing this!